crates/datafusion_ext/src/functions.rs

```diff
@@ -20,7 +20,11 @@ pub trait TableFunc: Sync + Send {
     /// function implementations.
     fn name(&self) -> &str;
     fn runtime_preference(&self) -> RuntimePreference;
-    fn detect_runtime(&self, _args: &[FuncParamValue]) -> Result<RuntimePreference> {
+    fn detect_runtime(
+        &self,
+        _args: &[FuncParamValue],
+        _parent: &RuntimePreference,
```

Comment:
```suggestion
        _parent: RuntimePreference,
```

Reply:
why would we want an owned version here?

Reply:
RuntimePreference is Copy (i32), so really comes down to passing a reference to an i32, or passing i32 itself. I normally just pass values when something is Copy.